Index: test/clicks_link_test.rb
===================================================================
--- test/clicks_link_test.rb	(revision 123)
+++ test/clicks_link_test.rb	(working copy)
@@ -159,4 +159,15 @@
     @session.expects(:get_via_redirect).with("/page1", {})
     @session.clicks_link "Link text"
   end
+  
+  def test_should_with_scoping
+    @response.stubs(:body).returns(<<-EOS)
+      <a href="/page1">Link text</a>
+      <div id="correct_link">
+        <a href="/page2">Link text</a>
+      </div>
+    EOS
+    @session.expects(:get_via_redirect).with("/page2", {})
+    @session.clicks_link "Link text", :scope => "#correct_link"
+  end
 end
Index: lib/webrat/session.rb
===================================================================
--- lib/webrat/session.rb	(revision 123)
+++ lib/webrat/session.rb	(working copy)
@@ -23,8 +23,9 @@
       #
       # Example:
       #   clicks_link "Sign up"
-      def clicks_link(link_text)
-        link = links.detect { |el| el.innerHTML =~ /#{link_text}/i }
+      def clicks_link(link_text, opts = {})
+        scope = opts[:scope]
+        link = links(scope).detect { |el| el.innerHTML =~ /#{link_text}/i }
         return flunk("No link with text #{link_text.inspect} was found") if link.nil?
         
         onclick = link.attributes["onclick"]
@@ -357,8 +358,12 @@
         @form_data ||= []
       end
       
-      def links # :nodoc:
-        (dom / "a[@href]")
+      def links(scope = nil) # :nodoc:
+        if scope
+          (dom / "#{scope}" / "a[@href]")
+        else
+          (dom / "a[@href]")
+        end
       end
       
       def form_number(form) # :nodoc:
